File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        start = 0
        end = 1
        maxProfit = 0
        size = len(prices)
        
        while end < size:
            if prices[start] > prices[end]:
                start = end
            else:
                curr = prices[end] - prices[start]
                maxProfit = max(maxProfit, curr)
            end += 1
        return maxProfit

File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int start = 0;
        int end = 1;
        int size = prices.size();
        int maxProfit = 0;
        int currProfit = 0;
            
        while(end < size){
            if(prices[start] > prices[end]){
                start = end;
            }
            else{
                currProfit = prices[end] - prices[start];
                maxProfit = max(maxProfit, currProfit);
            }
            end++;
        }
        return maxProfit;
            
        
    }
};